src/chapters/az-cli/walkthrough_az-cli.rst

```diff
@@ -0,0 +1,537 @@
+    $ keyvault_name="lc-20-<name>-kv"
+    $ az keyvault create -n "$keyvault_name"
+
+After the KeyVault has been provisioned let's set the connection string secret name and value:
+
+- ``--vault-name``: the KeyVault name [stored in a variable]
+- ``name``: "ConnectionStrings--Default"
+- ``value``: "server=localhost;port=3306;database=coding_events;user=coding_events;password=launchcode"
+
+.. admonition:: tip
+
+    Recall that secrets are like the other JSON entries in ``application.properties`` that we need to keep private and out of version control. The ``--`` is used as shorthand to define properties of JSON objects in a single "flat" string for the CLI command. In this case it is used to define a property called ``Default`` of a ``ConnectionStrings`` JSON object that would look like this:
+
+    .. sourcecode:: json
+
+        "ConnectionStrings": {
+            "Default": "<connection string value>"
+        }
+
+.. sourcecode:: powershell
+    :caption: Windows/PowerShell
+
+    > az keyvault secret set --vault-name "$KeyVaultName" -n "ConnectionStrings--Default" --value "server=localhost;port=3306;database=coding_events;user=coding_events;password=launchcode"
+
+.. sourcecode:: bash
+    :caption: Linux/BASH
+
+    $ az keyvault secret set --vault-name "$keyvault_name" -n "ConnectionStrings--Default" --value "server=localhost;port=3306;database=coding_events;user=coding_events;password=launchcode"
+
+Configuring
+-----------
+
+Now that the KeyVault and connection string secret have been managed, all that remains is to to set the access policy for the VM. Earlier we listed two arguments needed for the ``set-policy`` KeyVault command that whose values weren't immediately obvious, the ``--object-id`` and ``--secret-permissions``.
+
+Getting the VM Object ID
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+In order to grant access to a resource we need to provide a unique identifier for it. Earlier when we provisioned our VM we used the ``--assign-identity`` argument to generate and assign a **service principal identifier**. Azure documentation refers to this identifier as either a **principal ID** or an **object ID**. 
+
+The VM ``show`` command provided us with a JSON object of configuration details. Issue the ``show`` command again and look for the ``identity`` object property. Within this sub-object is the ``principalId`` that we need.
+
+We can capture this value in a variable by combining the VM ``show`` command with a ``--query`` filter:
+
+.. sourcecode:: powershell
+    :caption: Windows/PowerShell
+
+    > $VmObjectId="$(az vm show --query "identity.principalId")"
+
+.. sourcecode:: bash
+    :caption: Linux/BASH
+
+    $ vm_object_id="$(az vm show --query "identity.principalId" -o tsv)"
+
+.. tip::
+
+    While exploring the VM Group you may have noticed a Sub-Group called ``identity`` which is a shortcut for accessing the same information. How would you modify your command and ``--query`` to use this Sub-Group instead?
+
+Least-Privileged Access
```

Comment:
This section kind of comes out of nowhere. They get command after command after command. And then they hit this section and have to read a few paragraphs about least privileged access. I would recommend cutting this section down as much as possible to not take away from the walkthrough nature of this doc.